### PR TITLE
feat(gRPC): add tls configurations to the operator and helm

### PR DESCRIPTION
**Pull request by rksharma95**
**Purpose of PR?**:

Fixes #

**Does this PR introduce a breaking change?**

**If the changes in this PR are manually verified, list down the scenarios covered:**:

**Additional information for reviewer?** :
_Mention if this PR is part of any design or a continuation of previous PRs_

**Checklist:**
- [ ] Bug fix. Fixes #<issue number>
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [ ] Commit has unit tests
- [ ] Commit has integration tests

### CODE REVIEW

File: tests/go.mod [hunk 1]
```diff
@@ -1,16 +1,16 @@
 module github.com/kubearmor/KubeArmor/tests
 
-go 1.21
+go 1.22
```
Comment (rksharma95) [inline tests/go.mod]: kubearmor-client is using go1.22, so tests also. ig that's why snyk test failed. should I revert to go 1.21 for kubearmor-client or what?

Comment (DelusionalOptimist) [reply tests/go.mod]: Oof... we tried using go 1.22 in KubeArmor earlier but snyk was not working with it for some reason. We concluded that snyk didn't support it back then.
IG now support for 1.22 is there. So this should be fine.

Comment (DelusionalOptimist) [reply tests/go.mod]: @Aryan-sharma11 can you check why snyk (nyrahul) is failing?

Comment (Aryan-sharma11) [reply tests/go.mod]: ` Introduced through: github.com/kubearmor/KubeArmor/tests@0.0.0 › github.com/kubearmor/kubearmor-client@v1.2.0 › helm.sh/helm@v3.14.2`
but there's no version containing a fix for this vulnerability.

Comment (DelusionalOptimist) [reply tests/go.mod]: Gotcha. I think we can update go in "test" package for now, updates to other modules can be handled in a different PR.

File: deployments/helm/KubeArmor/values.yaml [hunk 2]
```diff
@@ -5,27 +5,63 @@ environment:
...
+tls:
+  enabled: false  
+
 kubearmorRelay:
   # to enable/disable kubearmor-relay
   enabled: true
   image:
     # kubearmor-init image repo
-    repository: kubearmor/kubearmor-relay-server
+    repository: rksharma95/kubearmor-relay-server
```
Comment (daemon1024) [inline deployments/helm/KubeArmor/values.yaml]: Need to update back, since PR is merged

**Review by daemon1024 — APPROVED** [review]

**Review by daemon1024 — CHANGES_REQUESTED** [review]
Tests seem to be passing, LGTM. let's just update the images to kubearmor official ones.

**Review by DelusionalOptimist — COMMENTED** [review]
@rksharma95 please rebase with main.

**Comment by rksharma95** [comment]
> @rksharma95 please rebase with main.

kubearmor-client pkg uses go1.22 and it will be an issue, i think we should use same go version there as well.

File: contribution/self-managed-k8s/crio/install_crio.sh [hunk 3]
```diff
@@ -12,21 +12,18 @@ fi
...
 
 # install cri-o corresponding to the latest k3s version
-VERSION=$(curl -w '%{url_effective}' -L -s -S https://update.k3s.io/v1-release/channels/stable -o /dev/null | sed -e 's|.*/v||' | cut -d '.' -f1,2)
+VERSION=$(curl -w '%{url_effective}' -L -s -S https://update.k3s.io/v1-release/channels/stable -o /dev/null | sed -e 's|.*/||' | cut -d '.' -f1,2)
 echo "Installing CRI-O version $VERSION"
 
-# get signing keys
-echo "deb [signed-by=/usr/share/keyrings/libcontainers-archive-keyring.gpg] https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/$OS/ /" | sudo tee /etc/apt/sources.list.d/devel:kubic:libcontainers:stable.list
-echo "deb [signed-by=/usr/share/keyrings/libcontainers-crio-archive-keyring.gpg] https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable:/cri-o:/$VERSION/$OS/ /" | sudo tee /etc/apt/sources.list.d/devel:kubic:libcontainers:stable:cri-o:$VERSION.list
+curl -fsSL https://pkgs.k8s.io/addons:/cri-o://stable:/$VERSION/deb/Release.key |
```
Comment (rksharma95) [inline contribution/self-managed-k8s/crio/install_crio.sh:18]: https://github.com/cri-o/packaging/blob/main/README.md

**Review by daemon1024 — DISMISSED** [review]

File: KubeArmor/cert/cert.go [hunk 4]
```diff
@@ -80,6 +80,19 @@ type CertPath struct {
...
 
+func GetPemCertFromx509Cert(cert x509.Certificate) []byte {
+	certPem := &pem.Block{
+		Type:  "CERTIFICATE",
+		Bytes: cert.Raw,
+	}
+	var pemBuffer bytes.Buffer
+	err := pem.Encode(&pemBuffer, certPem)
+	if err != nil {
+		fmt.Println("error while encoding certificate to pem format", err)
```
Comment (DelusionalOptimist) [inline KubeArmor/cert/cert.go]: Please return error or use logger.

File: KubeArmor/cert/cert.go [hunk 5]
```diff
@@ -186,6 +199,22 @@ func ReadCertFromK8sSecret(client *kubernetes.Clientset, namespace, secret strin
 	}, nil
 }
 
+func GenerateCA(cfg *CertConfig) (*CertBytes, error) {
+	crtTemp, err := GenerateCert(cfg)
+	if err != nil {
+		fmt.Printf("error generating ca cert: %s\n", err)
```
Comment (DelusionalOptimist) [inline KubeArmor/cert/cert.go]: Please return error or use logger.

**Review by DelusionalOptimist — COMMENTED** [review]
Looks good... Just some small changes to address.
Thanks.

**Review by daemon1024 — APPROVED** [review]

**Review by DelusionalOptimist — APPROVED** [review]
LGTM. Thanks @rksharma95.
Sorry for the late review!

**Pull request merged**